dash/dash.py

```diff
@@ -490,7 +491,17 @@ def layout(self):
         return self._layout
 
     def _layout_value(self):
-        return self._layout() if self._layout_is_function else self._layout
+        layout = self._layout() if self._layout_is_function else self._layout
+
+        # Add extra hidden components
+        if self._extra_components:
+            if not hasattr(layout, "children"):
```

Comment:
Some simple apps just use `app.layout = DataTable(...)` or a graph or something, that don't support children. Mostly this is just for testing purposes but maybe not always. In that case what do we do, throw an error? Wrap in `html.Div`? Actually there's something nice about wrapping in a div, then we don't need to keep testing if we've already added these components into the layout, and don't need to mutate it. I can't really think of how mutating the layout would cause a problem but it makes me uneasy.

Reply:
Can you think of any cases where wrapping in a Div would break something?  Any concern if the top-level is a `ddk.App` or `dbc.App`?

Reply:
I don't think this would cause a problem - we're already inside a div from the react entry point

https://github.com/plotly/dash/blob/a19c27426c7008769530d72420a88a68b9aed548/dash/dash.py#L86-L92

Could be nice at some point to allow the top level layout to be an array, but this would either also require wrapping in a div or would require changes to the renderer.

Reply:
Ok, wrapping in a div seems like the best solution then. Thanks!

Reply:
Done in b88d8a3